### ISSUE TITLE
123solar cant run (suddenly after a few months)

### ISSUE THREAD

**Issue by speeskeek** [jeanmarc77/123solar#62]
suddenly is maybe not fully true - I upgraded to newer debian - with php 8.2 - maybe that's the issue?

If I start 123solar on th e admin page (with the big button) most of the time it doesnt start. When I start at command line it starts, but it also stops somehow. I cant see or read why. even with debug on.

whenb running it has thje correct readings, and displays the value on the gauge.
but afetrer ! minute or so it stops...

Is ther eanywhere where I can see what is going wrong?
meterN is running fine.

[link to my site](https://bartman.duckdns.org/123solar/)

**Comment by jeanmarc77** [comment]
Hi,
Did you use github version or the release https://github.com/jeanmarc77/123solar/releases/tag/1.8.4.5 ?

Git is "work in progress" and use systemd to start/stop the script. It hasn't been tested thoroughly..

**Comment by speeskeek** [comment]
I reinstalled with the release (an not git). Same issue. Where can I look for a log file? Is it php log?

**Comment by jeanmarc77** [comment]
You can enable 'debug' then restart and check 123Solar errors log in help page.
Also you can have hints on php error log as well

**Comment by speeskeek** [comment]
Debug says nothing. 

`#* 10/01/2024 17:26:24	Starting 123Solar debug (3126995)`

it ran all night in debug without stopiing. Moment the sun comes out (and it has to do something) it stops.

**Comment by jeanmarc77** [comment]
and php error log ?
did you try the Enhanced com. tester in help page ?

**Comment by speeskeek** [comment]
![IMG_3677](https://github.com/jeanmarc77/123solar/assets/8826404/45b90e9b-021b-4792-84cd-407557aa026c)
![IMG_3680](https://github.com/jeanmarc77/123solar/assets/8826404/10721c36-affc-40f8-a42b-a5e8d2363fad)

**Comment by speeskeek** [comment]
> journalctl -u php-fpm -u php
$ sudo journalctl -u php-fpm -u php
-- No entries --



the other command (ournalctl -r | grep php) is not showing anything usefull now because its dark. I'll check the log again in the morning :)

**Comment by speeskeek** [comment]
root@Z800:~# journalctl -r | grep php
Jan 12 08:39:08 Z800 systemd[1]: Finished phpsessionclean.service - Clean php session files.
Jan 12 08:39:08 Z800 systemd[1]: phpsessionclean.service: Deactivated successfully.
Jan 12 08:39:07 Z800 systemd[1]: Starting phpsessionclean.service - Clean php session files...
Jan 12 08:39:01 Z800 CRON[3103893]: (root) CMD (  [ -x /usr/lib/php/sessionclean ] && if [ ! -d /run/systemd/system ]; then /usr/lib/php/sessionclean; fi)
Jan 12 08:15:13 Z800 php[3056871]: (standard_in) 1: syntax error
Jan 12 08:15:13 Z800 php[3056848]: parse error: Invalid numeric literal at line 1, column 7
Jan 12 08:15:12 Z800 php[3056752]: (standard_in) 1: syntax error
Jan 12 08:15:12 Z800 php[3056749]: parse error: Invalid numeric literal at line 1, column 7
Jan 12 08:15:11 Z800 php[3056729]: (standard_in) 1: syntax error



so it doesn't say something usefull arount the start or stop times of the script.

this is 123solar debug

20240112-08:57:02: aurora: No response after 3 attempts
20240112-08:57:02: aurora: ERROR: Received bad return code (-1 0)
20240112-08:57:02: aurora: No response after 3 attempts
20240112-08:57:02: aurora: ERROR: Received bad return code (-1 0)
20240112-08:57:02: aurora: No response after 3 attempts
20240112-08:57:02: aurora: ERROR: Received bad return code (-1 0)
#* 12/01/2024 09:06:36	Starting 123Solar debug (3168158)

#* 12/01/2024 09:06:59	Starting 123Solar debug (3168776)

#* 12/01/2024 09:10:45	Starting 123Solar debug (3175313)



but it has the aurora errors sometimes.

I'll try to replug the rs485 stick

**Comment by speeskeek** [comment]
can this be a problem ?
![Screenshot_20240112_090932](https://github.com/jeanmarc77/123solar/assets/8826404/2fff2182-ee32-43dd-bd94-c6c1303cec0d)

**Comment by jeanmarc77** [comment]
don't start with command line and make sure there isn't a 123solar.pid file.
Check also what php scripts runs

**Comment by speeskeek** [comment]
Here are the processes. I couldnt find an old PID file. 
it starts but after one minute it stops 

is it normale the pid’s of the aurora commands are changing?

![IMG_3717](https://github.com/jeanmarc77/123solar/assets/8826404/d1b06305-56dc-4228-b60b-c87d9e88835e)
![IMG_3718](https://github.com/jeanmarc77/123solar/assets/8826404/98dcdfe4-ec42-433a-b2fa-6631fa869944)
![IMG_3719](https://github.com/jeanmarc77/123solar/assets/8826404/fd1aa8d1-1779-4eee-adc8-53495e1c834b)
![IMG_3720](https://github.com/jeanmarc77/123solar/assets/8826404/31d5e62e-40c5-41dd-a85a-d4e5ed8bfbc7)
![IMG_3721](https://github.com/jeanmarc77/123solar/assets/8826404/f02c0618-1bba-4f0a-8b82-d8035245726d)
![IMG_3722](https://github.com/jeanmarc77/123solar/assets/8826404/14855d18-e856-4f97-a7af-388f5f200d3f)
![IMG_3723](https://github.com/jeanmarc77/123solar/assets/8826404/9b4cc037-f2f4-473f-be13-245a44a8f76d)

**Comment by jeanmarc77** [comment]
Yes, it's normal pid is changing continuously.
Which [123solar.service] do you use ?

**Comment by speeskeek** [comment]
> 123solar.service]

Before all the trouble I was running it wirh a service. Since the problems started I was using not the service - I was starting from the web interface.

Now I started it again with service. It runs. but fails to - but since it's a service it restarts... so I have my logging bacl - but it's not okay to have these errors?


`bart@Z800:~$ sudo journalctl --follow -u 123solar.service
Jan 14 11:57:19 Z800 systemd[1]: Starting 123solar.service - 123Solar...
Jan 14 11:57:19 Z800 systemd[1]: Started 123solar.service - 123Solar.
Jan 14 12:06:00 Z800 systemd[1]: 123solar.service: Main process exited, code=exited, status=255/EXCEPTION
Jan 14 12:06:00 Z800 systemd[1]: 123solar.service: Failed with result 'exit-code'.
Jan 14 12:06:00 Z800 systemd[1]: 123solar.service: Consumed 10.136s CPU time.
Jan 14 12:06:30 Z800 systemd[1]: 123solar.service: Scheduled restart job, restart counter is at 1.
Jan 14 12:06:30 Z800 systemd[1]: Stopped 123solar.service - 123Solar.
Jan 14 12:06:30 Z800 systemd[1]: 123solar.service: Consumed 10.136s CPU time.
Jan 14 12:06:30 Z800 systemd[1]: Starting 123solar.service - 123Solar...
Jan 14 12:06:30 Z800 systemd[1]: Started 123solar.service - 123Solar.
Jan 14 12:06:31 Z800 systemd[1]: 123solar.service: Main process exited, code=exited, status=255/EXCEPTION
Jan 14 12:06:31 Z800 systemd[1]: 123solar.service: Failed with result 'exit-code'.
Jan 14 12:07:01 Z800 systemd[1]: 123solar.service: Scheduled restart job, restart counter is at 2.
Jan 14 12:07:01 Z800 systemd[1]: Stopped 123solar.service - 123Solar.
Jan 14 12:07:01 Z800 systemd[1]: Starting 123solar.service - 123Solar...
Jan 14 12:07:01 Z800 systemd[1]: Started 123solar.service - 123Solar.
Jan 14 12:07:02 Z800 systemd[1]: 123solar.service: Main process exited, code=exited, status=255/EXCEPTION
Jan 14 12:07:02 Z800 systemd[1]: 123solar.service: Failed with result 'exit-code'.
Jan 14 12:07:32 Z800 systemd[1]: 123solar.service: Scheduled restart job, restart counter is at 3.
Jan 14 12:07:32 Z800 systemd[1]: Stopped 123solar.service - 123Solar.
Jan 14 12:07:32 Z800 systemd[1]: Starting 123solar.service - 123Solar...
Jan 14 12:07:32 Z800 systemd[1]: Started 123solar.service - 123Solar.
Jan 14 12:07:33 Z800 systemd[1]: 123solar.service: Main process exited, code=exited, status=255/EXCEPTION`

**Comment by speeskeek** [comment]
> Which [123solar.service] do you use ?

> bart@Z800:~$ cat /etc/systemd/system/123solar.service                                                                  
[Unit]                                                                                                                 
Description=123Solar                                                                                                   
Requires=network.target                                                                                                
After=network.target nginx.service php-fpm.service                                                                     
StartLimitIntervalSec=60                                                                                               
StartLimitBurst=5
 
[Service]
Type=simple
WorkingDirectory=/var/www/bartman/123solar/scripts/
ExecStart=/usr/bin/php 123solar.php
# force PIDFILE see 'man systemd.service'
ExecStartPost=/bin/sh -c 'systemctl show -p MainPID --value 123solar.service > /var/www/bartman/123solar/scripts/123solar.pid'
ExecStopPost=/usr/bin/rm -f /var/www/bartman/123solar/scripts/123solar.pid
User=www-data
Group=www-data
Restart=always
RestartSec=30s
#Since version 7.4 php-fpm is hardened
ReadWritePaths = /var/lock
 
[Install]
WantedBy=default.target

**Comment by speeskeek** [comment]
not great .. because it  is not loggin all :
(it is 14:00 now)
![Uploading Screenshot_20240114_140017.png…]()

**Comment by speeskeek** [comment]
not working great .. because it is not loggin all :
(it is 14:00 now)

![Screenshot_20240114_140017](https://github.com/jeanmarc77/123solar/assets/8826404/a885bf98-a060-45aa-b5dd-9c4dd2310e38)

**Comment by jeanmarc77** [comment]
Hi,
You rather should use the old [service](https://github.com/jeanmarc77/123solar/blob/main/misc/examples/123solar_old.service) 
The new one respawn but it haven't tested thoroughly.

It's odd you don't have info in logs files (php and 123s)
Check what's running before and after it hang with "ps -ef | grep php"

**Comment by speeskeek** [comment]
I tried the old service...
bot very mucht running after starting it

> udo systemctl status 123solar_old.service 
○ 123solar_old.service - 123Solar
     Loaded: loaded (/etc/systemd/system/123solar_old.service; enabled; preset: enabled)
     Active: inactive (dead) since Sun 2024-01-14 14:44:07 CET; 2s ago
    Process: 1539896 ExecStart=/usr/bin/curl http://bartman.duckdns.org/123solar/scripts/boot123s.php (code=exited, status=0/SUCCESS)
   Main PID: 1539896 (code=exited, status=0/SUCCESS)
        CPU: 31ms

Jan 14 14:44:07 Z800 systemd[1]: Starting 123solar_old.service - 123Solar...
Jan 14 14:44:07 Z800 curl[1539896]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
Jan 14 14:44:07 Z800 curl[1539896]:                                  Dload  Upload   Total   Spent    Left  Speed
Jan 14 14:44:07 Z800 curl[1539896]: [158B blob data]
Jan 14 14:44:07 Z800 curl[1539896]: Direct access not permitted
Jan 14 14:44:07 Z800 systemd[1]: 123solar_old.service: Deactivated successfully.
Jan 14 14:44:07 Z800 systemd[1]: Finished 123solar_old.service - 123Solar.



secondly I tried running the boot123.php directly, but that wasnt allowed

> sudo -u www-data php scripts/boot123s.php 
PHP Warning:  Undefined array key "SERVER_ADDR" in /var/www/bartman/123solar/scripts/boot123s.php on line 9
PHP Warning:  Undefined array key "SERVER_ADDR" in /var/www/bartman/123solar/scripts/boot123s.php on line 9
Direct access not permitted


So I modified the script for trying:

I commented out 

> #if ($_SERVER['SERVER_ADDR'] != '127.0.0.1' && $_SERVER['SERVER_ADDR'] != '::1') {
#       die('Direct access not permitted');
#}


then it ran for 20secs... so no change


(now I uncommented the lines again)

**Comment by jeanmarc77** [comment]
Well, give up the .service. 
Something wrong as your data/123solar.err file don't write today details

**Comment by speeskeek** [comment]
Dont know how it’s running now - but it runs since yesterday evening.  Ain’t gonna touch it. 

When browsing trough the files I saw some files weren’t owned by www-data. I think that fixed it. 
but that was one of the first things I did (sudo chown -R www-data * ).  Also the file permission checker didn’t remark anything. Happy for now. Thanks dor the answers and support. 

Luckily these were dark days/weeks so not a lot of logging is lost

**Comment by speeskeek** [comment]
I was too enthousiast… It stops right after starting (on the webpage)

**Comment by jeanmarc77** [comment]
Sorry but i cannot debug without error log files. Try to give more details..

**Comment by speeskeek** [comment]
I don’t know what log to post here but….
hey … good news. It is working stable since a week now (in debug). I did a reboot (unintentionally) and after the reeboot it was still working (in debug). I disabled debug just now and it seems to be working. 
I disabled the pv output.. maybe there was some error there. 
not sure if I’ll enable it (there is this nice app PVoutputPro op iOS that is nice to use) later again
